src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Properties.jl

```diff
@@ -566,23 +566,16 @@ end
   return true
 end
 
-@attr Bool function is_reduced(X::AbsSpec{<:Field, <:MPAnyNonQuoRing})
-  return true
-end
-
 @attr Bool function is_geometrically_reduced(X::AbsSpec{<:Field, <:MPAnyNonQuoRing})
   return true
 end
 
-function is_perfect(F::Field)
-  characteristic(F) == 0 && return true
-  absolute_degree(F) < inf && return true
-  throw(NotImplementedError())
-end
-
 @attr Bool function is_geometrically_reduced(X::AbsSpec{<:Field, <:MPAnyQuoRing})
   F = base_ring(X)
-  is_perfect(F) && return is_reduced(X)
+  # is_perfect(F) && return is_reduced(X)
+  if characteristic(F) == 0 || F isa FinField
+    return true
+  end
   throw(NotImplementedError())
 end
```

Comment:
I understand, why you removed 'is_perfect' further above -- is can only certify true in certain classes of cases, but a 'no' does not say anything about being perfect or not.

But here you only check whether F is certified to be perfect now, not whether it is_reduced, but you intend to determine is_geometrically_reduced.

Reply:
Right, I forgot.  `is_perfect` moves to `AbstractAlgebra`